src/sqlfluff/rules/L009.py

```diff
@@ -8,7 +8,59 @@
 
 @document_fix_compatible
 class Rule_L009(BaseRule):
-    """Files must end with a trailing newline."""
+    """Files must end with a trailing newline.
+
+    | **Anti-pattern**
+    | The content in file without ends without a trailing newline, the $ represents end of file.
+
+    .. code-block:: sql
+
+        SELECT
+            a
+        FROM foo$
+        
+        -- Ending on an indented line means there is no newline at the end of the file, the • represents space.
+
+        SELECT
+        ••••a
+        FROM
+        ••••foo
+        ••••$
+
+        -- Ending on a semi-colon means the last line is not a newline.
+
+        SELECT
+            a
+        FROM foo
+        ;$
+
+    | **Best practice**
+    | Add trailing newline to the end, the $ character represents end of file.
+
+    .. code-block:: sql
+
+        SELECT
+            a
+        FROM foo
+        $
+        
+        -- Ensuring the last line is not indented so is just a newline.
+
+       SELECT
```

Comment:
Missing a white space
```suggestion
        SELECT
```

Reply:
Fix it